### PR TITLE
add parentheses while pretty printing MatMul and KroneckerProduct

### LINKED ISSUE #15560: Missing parentheses pretty printing MatMul and KroneckerProduct
It seems this case wasn't fixed with https://github.com/sympy/sympy/issues/15439

```py
>>> i, j = symbols('i j', integer=True)
>>> X = MatrixSymbol('X', 1, j*k)
>>> A1 = MatrixSymbol("A1", j, 1)
>>> A2 = MatrixSymbol("A2", k, 1)
>>> pprint(X*KroneckerProduct(A1, A2))
X⋅A₁ ⨂ A₂
```

The other way is parenthesized

```py
>>> B = MatrixSymbol('B', i, j)
>>> pprint(KroneckerProduct(B*A1, A2))
(B⋅A₁) ⨂ A₂
```

But there is no implied precedence between a matrix product and Kronecker product, so both ways should be parenthesized.

### PR DESCRIPTION
**Pull request by rational-kunal**
#### References to other Issues or PRs
"Fixes #15560"

#### Brief description of what is fixed or changed
Modified `_print_MatMul` such that if matrix multiplication contains `KroneckerProduct` then it would add parentheses to it.
for example
`pprint(a*(KroneckerProduct(b, c)))`
`a⋅(b ⨂ c)`

#### Release Notes

- printing
  - MatMuls with KroneckerProducts now parenthesize correctly

### CODE REVIEW

File: examples/beginner/print_pretty.py [hunk 1]
```diff
@@ -39,5 +43,8 @@ def main():
     pprint( (1/cos(x)).series(x, 0, 10) )
     print('\n')
 
+    pprint(a*(KroneckerProduct(b, c)))
```
Comment (asmeurer) [inline examples/beginner/print_pretty.py]: Modifying the examples is fine, but the test for this needs to go in the test file (sympy/printing/pretty/tests/test_pretty.py).

**Comment by asmeurer** [comment]
Release notes should be written with the end user in mind. I've edited it to something more appropriate.

File: examples/beginner/print_pretty.py (hunk 1 above)
Comment (rationa-kunal) [reply examples/beginner/print_pretty.py]: I have added tests for this issue.
thanks for understanding noob

File: sympy/printing/pretty/tests/test_pretty.py [hunk 2]
```diff
@@ -6476,3 +6476,18 @@ def test_pretty_print_tensor_partial_deriv():
...
+    import sys
+    from sympy.core.compatibility import StringIO
+    fd = StringIO()
+    a = MatrixSymbol("a", 1, 1)
+    sso = sys.stdout
+    sys.stdout = fd
+    try:
+        pprint(a*(KroneckerProduct(a, a)), use_unicode=False, wrap_line=False)
+    finally:
+        sys.stdout = sso
```
Comment (asmeurer) [inline sympy/printing/pretty/tests/test_pretty.py]: Write the test like the other tests in the file. There is no need to capture stdout.

Comment (rationa-kunal) [reply sympy/printing/pretty/tests/test_pretty.py]: Updated the function ... 
Thanks

**Pull request merged**